src/handlers/int-test-support/helpers/commonHelpers.ts

```diff
@@ -57,15 +57,26 @@ export const waitForTrue = async (
 export const poll = async <Resolution>(
   promise: () => Promise<Resolution>,
   completionCondition: (resolution: Resolution) => boolean,
-  interval: number = 1_000,
-  timeout: number = 30_000
-): Promise<Resolution> =>
-  await new Promise((resolve, reject) => {
+  options?: {
+    interval?: number;
+    timeout?: number;
+    nonCompleteErrorMessage?: string;
+  }
+): Promise<Resolution> => {
+  const {
+    interval = 1_000,
+    timeout = 30_000,
+    nonCompleteErrorMessage = "Polling completion condition was never achieved",
+  } = options ?? {};
+  return await new Promise((resolve, reject) => {
     // This timeout safely exits the function if the completion condition
     // isn't achieved within the given timeout
     const timeoutHandle = setTimeout(() => {
       clearInterval(intervalHandle);
-      reject(new Error("Polling completion condition was never achieved"));
+      // Rejecting with a string rather than an error so that the failure
+      // bubbles up to the test, giving better output
```

Comment:
Yup. So in the previous version where we did something like `reject(new Error('message'))` this made the test failure point to the place in the code where that new Error was, which was always inside this `poll` function. That's not that useful because some of the tests use multiple polls so you couldn't actually tell what part of the test failed. With this change the error is actually the uncaught rejection in the test so that's what you'll see in the output.

Reply:
I'll break one so you can see the difference hold on

Reply:
```
const rejectWithError = async (): Promise<void> =>
  await new Promise((_resolve, reject) => {
    reject(new Error("Oops"));
  });

const rejectWithString = async (): Promise<void> =>
  await new Promise((_resolve, reject) => {
    // eslint-disable-next-line prefer-promise-reject-errors
    reject("Oops");
  });

describe("Rejection tests", () => {
  test("reject with error", async () => {
    await rejectWithError();
  });
  test("reject with string", async () => {
    await rejectWithString();
  });
});
```

I wrote this lil model, when I ran these this is the output

<img width="702" alt="Screenshot 2023-02-16 at 15 27 05" src="https://user-images.githubusercontent.com/35689158/219411039-7118f853-16a0-47d9-841b-4aa33d5e26aa.png">